app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/ReviewListFragment.kt

```diff
@@ -0,0 +1,89 @@
+package com.github.sdp.ratemyepfl.fragment.review
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.View
+import androidx.fragment.app.activityViewModels
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.RecyclerView
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.activity.AddReviewActivity
+import com.github.sdp.ratemyepfl.adapter.ReviewAdapter
+import com.github.sdp.ratemyepfl.model.items.Reviewable
+import com.github.sdp.ratemyepfl.model.review.Review
+import com.github.sdp.ratemyepfl.utils.ListActivityUtils
+import com.github.sdp.ratemyepfl.viewmodel.ReviewViewModel
+import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
+
+open class ReviewListFragment : Fragment(R.layout.fragment_review_list) {
+    protected lateinit var reviewsAdapter: ReviewAdapter
+    protected lateinit var recyclerView: RecyclerView
+
+    protected lateinit var fab: ExtendedFloatingActionButton
+    protected lateinit var swipeRefresher: SwipeRefreshLayout
+
+    private val viewModel by activityViewModels<ReviewViewModel>()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // Floating action button for adding reviews
+        fab = view.findViewById(R.id.startReviewFAB)
+
+        // Vertical swipe refreshes the list of reviews
+        swipeRefresher = view.findViewById(R.id.reviewSwipeRefresh)
+
+        // List of reviews
+        recyclerView = view.findViewById(R.id.reviewRecyclerView)
+
+        reviewsAdapter = ReviewAdapter()
+        recyclerView.adapter = reviewsAdapter
+
+        recyclerView.addItemDecoration(
+            DividerItemDecoration(view.context, DividerItemDecoration.VERTICAL)
+        )
+
+        recyclerView.setOnScrollListener(
+            ListActivityUtils.createOnScrollListener(
+                { fab.extend() },
+                { fab.shrink() }
+            )
+        )
+
+        swipeRefresher.setOnRefreshListener {
+            viewModel.updateReviewsList()
+            swipeRefresher.isRefreshing = false
+        }
+
+        setUpObservers()
+    }
+
+    fun setUpObservers(){
+        viewModel.getReviews().observe(viewLifecycleOwner) {
+            it?.let {
+                reviewsAdapter.submitList(it as MutableList<Review>)
```

Comment:
Here happens the same scheme. With a List<Review>, we could call it.toMutableList() instead of a cast.

Reply:
Not my code :)

Reply:
I will anyway add what you proposed,